### DISCUSSION TITLE
Charges Dashboard: Not showing KM driven for last charge

### DISCUSSION THREAD

**Discussion by coryschak** [teslamate-org/teslamate#2722] [General]
On the Charges Dashoard of Grafana, when you filter the results it always shows the last charge without any KM driven. Is there a way to fix this so it shows KM driven for all charges? Please see screenshot below:

![image](https://user-images.githubusercontent.com/24301280/175848584-0c1f22c8-3c13-4f78-a802-94321f1a3f88.png)

**Comment by tobiasehlert** [comment]
@coryschak, is this a fresh installation of TeslaMate?

**Reply by cwanja** [reply]
It is happening on my instance that has been running for over six months. I have not had a chance to look into it, but it definitely seems a bug.

In this image, where the `Geofence` is set to All, data displays. 
![Screen Shot 2022-06-29 at 10 00 08 AM](https://user-images.githubusercontent.com/54645561/176469638-6e5df0e9-0001-417b-8cca-28b6c36863d1.png)

As soon as I filter on the Supercharger geofence, it drops the data for the last row. The data is clearly there, from the image above. But it gets dropped upon filter.
![Screen Shot 2022-06-29 at 10 00 18 AM](https://user-images.githubusercontent.com/54645561/176469667-d24554ec-5c36-4c5e-8048-a6cbde599bb5.png)

**Reply by coryschak** [reply]
> @coryschak, is this a fresh installation of TeslaMate?

Pretty fresh, installed mid March - Version 1.26.1

**Reply by tobiasehlert** [reply]
@cwanja, I have made some testing on this and to me it looks like this.

The Driven column is calculated based on the odometer difference from the line below. Therefore the last line will always be empty, independent if you are running a window of 7 days or 2 years.

If you are for example filtering on your home-location only, where you are probably are charging the most. You will notice that some charges can have lot higher _driven_ values than the range your car has.

I for example have this when filtering 90 days on a supercharger near Stockholm (where I occasionally are charging when visiting relatives). And I do drive a MS 75D and have definitely not 877 km of charge in my car or even more as shown on the next line.. and then of course the empty line at the end :)

![image](https://user-images.githubusercontent.com/4417850/176536899-322661a8-2eb8-43d1-a018-f5ede47594ea.png)

So to summon this.. the driven column contains the traveled distance from the last charge shown in the filtered/selected view and not since the actual charge happening before.

@coryschak, alright thanks for the answer. Hope the answer above explains why you see this.

**Reply by cwanja** [reply]
That is a good catch @tobiasehlert - never put that together. Though the Driven column seems, for me at least on miles, is mostly true and accurate.

**Reply by coryschak** [reply]
> 

Makes sense, for the most part the driven distance is accurate for me unless you start filtering the results based on location, i always look at X days ago so this doesn't really affect me but i didn't notice that till you mentioned it.

I'm just using the Driven distance in some calculations for cost analysis and thats why i noticed blank entry. Won't kill me :)

What i been working on, i think the alterations i have are more geared towards my interests but willing to share if anyone wants. The values are a bit off since the last charge has a NULL driving distance so it throws some calcs off

![image](https://user-images.githubusercontent.com/24301280/176684296-ee21d75f-8047-4d26-a0fc-6a77a8d25ea1.png)
